### PR TITLE
cleaned up a bit the main process

### PR DESCRIPTION
**Pull request by david-andrew**
- handled case where `supporting docs` was not supplied
- replaced `input` with `easyrepl.readl` for reading input with better features, such as ability moving the cursor
    - adds `easyrepl` library
- fixed bug where only an empty set would be shown to the agent when it downloaded files
- moved some stuff around to be more logically organized

### CODE REVIEW

File: jvoy/run.py [hunk 1]
```diff
@@ -124,22 +124,16 @@ def main():
...
+                            Role.system,
+                            content="The page has changed since the decision was made. Please retry making an new decision with the same task.",
+                        ))
                         continue
                     
                     #check to see if file is downloaded
                     new_downloads = download_tracker.get_new_downloads() 
                     if len(new_downloads) > 0: 
-                        file_obs = f"You downloaded files named {download_tracker.get_new_downloads()}. This may mean that your task is complete, evaluate and proceed accordingly."
+                        file_obs = f"You downloaded files named {new_downloads}. This may mean that your task is complete, evaluate and proceed accordingly."
```
Comment (ryanholtschneider2) [inline jvoy/run.py:136]: This seems fine for now but note that any interactions with the downloaded file are likely to be fragile. 

When you click a download button there is likely to be an intermediary file placeholder for the currently downloading file if the file download is longer than the click sleep time.

If in future we replace or reduce the click sleep time or want to use the downloaded doc we need to make sure we get the proper name of the file and wait long enough for the download

**Review by ryanholtschneider2 — APPROVED** [review]
Looks good to me. Just left one comment as a reminder for future use of downloaded files. 
We need to handle interacting with the web page dynamically in general, right now we rely on some manually set sleep timers (likely overly long).

Nevermind, one minor issue, see below

File: jvoy/run.py [hunk 2]
```diff
@@ -32,55 +32,55 @@ def main():
...
     driver_task.get(args.start_page)
-    
-    #add extra information related to the task/site
-    supporting_message=build_supporting_docs_message(args.supporting_docs)
-    logger.log_message(f"## Supporting Information Task:\n{supporting_message}")
-    # get the user task
-    user_task = input("Tell the LLM agent what to do: ")
-    logger.log_message(f"## User Task:\n{user_task}")
+
+    # set up the download tracker
```
Comment (ryanholtschneider2) [inline jvoy/run.py:50]: This needs to get moved back up. It is below where it is used in setting up the selenium driver.

**Review by ryanholtschneider2 — CHANGES_REQUESTED** [review]
move download logic back above where it is used

File: jvoy/run.py (hunk 1 above)
Comment (david-andrew) [reply jvoy/run.py:136]: yeah I agree, I think a big thing we need to work on at some point will be better waiting. So when downloading stuff, the process should wait till the download finishes, and in general, it should wait till a page is settled before it has the agent do anything

File: jvoy/run.py (hunk 2 above)
Comment (david-andrew) [reply jvoy/run.py:50]: ah yeah my bad, let me adjust that

**Review by ryanholtschneider2 — APPROVED** [review]

**Pull request merged**